packages/dates/src/utilities/formatDate.ts

```diff
@@ -12,9 +12,11 @@ export function memoizedGetDateTimeFormat(
   return i;
 }
 
-const browserFeatureDetectionDate = Intl.DateTimeFormat('en', {
-  hour: 'numeric',
-});
+const browserFeatureDetectionDate = (options: FormatDateOptions) =>
+  Intl.DateTimeFormat('en', {
```

Comment:
Why does this need to be changed? 

I feel pretty certain that `Intl.DateTimeFormat('en', {hour: 'numeric'}).resolvedOptions()` would always result in a gregorian calendar. We never call `.format` on the result of this, so it seems like it wouldn't ever be an issue in the Thai locale.


If my expectation is right and this is not needed, then nothing in this file needs to be changed.